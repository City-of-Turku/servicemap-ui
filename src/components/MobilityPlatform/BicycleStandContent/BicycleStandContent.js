import React from 'react';
import PropTypes from 'prop-types';
import { Typography } from '@material-ui/core';

const BicycleStandContent = ({
  classes, standName, standModel, standCover, hullLockable, numOfPlaces, numOfStands, intl,
}) => {
  const titleTypo = (props = {}) => (
    <div className={classes.title}>
      <Typography variant="subtitle1" {...props}>
        {standName}
      </Typography>
    </div>
  );

  const multiValueTypo = (
    message1Id,
    message2Id,
    message3Id,
    message4Id,
    message5Id,
    message6Id,
    message7Id,
  ) => (
    <div className={classes.content}>
      <Typography variant="body2" display="block">
        <strong>
          {intl.formatMessage({
            id: message1Id,
          })}
          :
        </strong>
        {' '}
        {standModel}
      </Typography>
      <Typography variant="body2" display="block">
        <strong>
          {intl.formatMessage({
            id: message2Id,
          })}
          :
        </strong>
        {' '}
        {numOfPlaces}
      </Typography>
      <Typography variant="body2" display="block">
        <strong>
          {intl.formatMessage({
            id: message3Id,
          })}
          :
        </strong>
        {' '}
        {numOfStands}
      </Typography>
      {standCover ? (
        <Typography variant="body2" display="block">
          {intl.formatMessage({
            id: message4Id,
          })}
        </Typography>
      ) : (
        <Typography variant="body2" display="block">
          {intl.formatMessage({
            id: message5Id,
          })}
        </Typography>
      )}
      {hullLockable ? (
        <Typography variant="body2" display="block">
          {intl.formatMessage({
            id: message6Id,
          })}
        </Typography>
      ) : (
        <Typography variant="body2" display="block">
          {intl.formatMessage({
            id: message7Id,
          })}
        </Typography>
      )}
    </div>
  );

  return (
    <div>
      {titleTypo()}
      {multiValueTypo(
        'mobilityPlatform.content.bicycleStands.model',
        'mobilityPlatform.content.bicycleStands.numOfPlaces',
        'mobilityPlatform.content.bicycleStands.numOfStands',
        'mobilityPlatform.content.bicycleStands.covered',
        'mobilityPlatform.content.bicycleStands.notCovered',
        'mobilityPlatform.content.bicycleStands.hullLockable',
        'mobilityPlatform.content.bicycleStands.hullNotLockable',
      )}
    </div>
  );
};

BicycleStandContent.propTypes = {
  classes: PropTypes.objectOf(PropTypes.any).isRequired,
  standName: PropTypes.string,
  standModel: PropTypes.string,
  standCover: PropTypes.bool,
  hullLockable: PropTypes.bool,
  numOfPlaces: PropTypes.number,
  numOfStands: PropTypes.number,
  intl: PropTypes.objectOf(PropTypes.any).isRequired,
};

BicycleStandContent.defaultProps = {
  standName: '',
  standModel: '',
  standCover: false,
  hullLockable: false,
  numOfPlaces: 0,
  numOfStands: 0,
};

export default BicycleStandContent;
