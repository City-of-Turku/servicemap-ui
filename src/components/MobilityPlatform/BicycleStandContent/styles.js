const styles = {
  title: {
    marginBottom: '0.5rem',
    width: '85%',
    borderBottom: '1px solid #000000',
  },
};

export default styles;
