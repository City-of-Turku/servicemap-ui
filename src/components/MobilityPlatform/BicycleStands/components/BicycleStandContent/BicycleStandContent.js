import { Typography } from '@mui/material';
import PropTypes from 'prop-types';
import React from 'react';
import { useIntl } from 'react-intl';
import { StyledContainer, StyledHeaderContainer, StyledTextContainer } from '../../../styled/styled';

const BicycleStandContent = ({ bicycleStand }) => {
  const intl = useIntl();

  const messageIds = {
    model: 'mobilityPlatform.content.bicycleStands.model',
    places: 'mobilityPlatform.content.bicycleStands.numOfPlaces',
    stands: 'mobilityPlatform.content.bicycleStands.numOfStands',
    covered: 'mobilityPlatform.content.bicycleStands.covered',
    notCovered: 'mobilityPlatform.content.bicycleStands.notCovered',
    lockable: 'mobilityPlatform.content.bicycleStands.hullLockable',
    notLockable: 'mobilityPlatform.content.bicycleStands.hullNotLockable',
    maintained: 'mobilityPlatform.content.bicycleStands.maintainedByTku',
  };

  const titleTypo = () => (
    <StyledHeaderContainer>
      <Typography variant="subtitle1" component="h3">
        {bicycleStand.name}
      </Typography>
    </StyledHeaderContainer>
  );

  const multiValueTypo = messages => (
    <div>
      {bicycleStand.extra.model ? (
        <StyledTextContainer>
          <Typography component="p" variant="body2">
            {intl.formatMessage({
              id: messages.model,
            }, { value: bicycleStand.extra.model })}
          </Typography>
        </StyledTextContainer>
      ) : null}
      <StyledTextContainer>
        <Typography component="p" variant="body2">
          {intl.formatMessage({
            id: messages.places,
          }, { value: bicycleStand.extra.number_of_places })}
        </Typography>
      </StyledTextContainer>
      <StyledTextContainer>
        <Typography component="p" variant="body2">
          {intl.formatMessage({
            id: messages.stands,
          }, { value: bicycleStand.extra.number_of_stands })}
        </Typography>
      </StyledTextContainer>
      {bicycleStand.extra.covered ? (
        <StyledTextContainer>
          <Typography component="p" variant="body2" display="block">
            {intl.formatMessage({
              id: messages.covered,
            })}
          </Typography>
        </StyledTextContainer>
      ) : (
        <StyledTextContainer>
          <Typography component="p" variant="body2" display="block">
            {intl.formatMessage({
              id: messages.notCovered,
            })}
          </Typography>
        </StyledTextContainer>
      )}
      {bicycleStand.extra.hull_lockable ? (
        <StyledTextContainer>
          <Typography component="p" variant="body2" display="block">
            {intl.formatMessage({
              id: messages.lockable,
            })}
          </Typography>
        </StyledTextContainer>
      ) : (
        <StyledTextContainer>
          <Typography component="p" variant="body2" display="block">
            {intl.formatMessage({
              id: messages.notLockable,
            })}
          </Typography>
        </StyledTextContainer>
      )}
      {bicycleStand.extra.maintained_by_turku ? (
        <StyledTextContainer>
          <Typography component="p" variant="body2" display="block">
            {intl.formatMessage({
              id: messages.maintained,
            })}
          </Typography>
        </StyledTextContainer>
      ) : null}
    </div>
  );

  return (
    <StyledContainer>
      {titleTypo()}
      {multiValueTypo(messageIds)}
    </StyledContainer>
  );
};

BicycleStandContent.propTypes = {
  bicycleStand: PropTypes.shape({
    name: PropTypes.string,
    extra: PropTypes.shape({
      model: PropTypes.string,
      number_of_places: PropTypes.number,
      number_of_stands: PropTypes.number,
      covered: PropTypes.bool,
      hull_lockable: PropTypes.bool,
      maintained_by_turku: PropTypes.bool,
    }),
  }),
};

BicycleStandContent.defaultProps = {
  bicycleStand: {},
};

export default BicycleStandContent;
