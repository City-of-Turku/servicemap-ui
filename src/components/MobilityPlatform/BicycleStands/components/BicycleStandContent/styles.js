export default theme => ({
  title: {
    marginBottom: theme.spacing(1),
    width: '85%',
    borderBottom: '1px solid #000000',
  },
  titleText: {
    fontSize: '0.9rem',
  },
  paragraph: {
    marginBottom: theme.spacing(0.5),
  },
  padding: {
    padding: theme.spacing(1),
  },
});
