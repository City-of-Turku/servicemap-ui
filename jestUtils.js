import React from 'react';
import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { ThemeProvider } from '@mui/material/styles';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import themes from './src/themes';
import finnish from './src/i18n/fi';

const mockStore = configureStore([]);

// Mock props for intl provider
const intlMock = {
  locale: 'fi',
  messages: finnish,
  wrapRichTextChunksInFragment: false,
};

const Providers = (mockState) => ({ children }) => {
  const store = mockStore(mockState);
  return (
    <Provider store={store}>
      <IntlProvider {...intlMock}>
        <ThemeProvider theme={themes.SMTheme}>
          {children}
        </ThemeProvider>
      </IntlProvider>
    </Provider>
  );
}

export const getRenderWithProviders = mockState => component => render(component, { wrapper: Providers(mockState) });