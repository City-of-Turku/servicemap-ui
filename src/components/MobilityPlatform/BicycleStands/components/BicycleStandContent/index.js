import { withStyles } from '@mui/styles';
import { injectIntl } from 'react-intl';
import BicycleStandContent from './BicycleStandContent';
import styles from './styles';

export default withStyles(styles)(injectIntl(BicycleStandContent));
