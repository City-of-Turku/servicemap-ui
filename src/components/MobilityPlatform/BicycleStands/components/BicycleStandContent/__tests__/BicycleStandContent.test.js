// Link.react.test.js
import React from 'react';
import BicycleStandContent from '../index';
import finnishTranslations from '../../../../../../i18n/fi';
import { getRenderWithProviders } from '../../../../../../../jestUtils';

const mockProps = {
  bicycleStand: {
    name: 'Testinimi',
    extra: {
      model: 'Testimalli',
      number_of_places: 10,
      number_of_stands: 5,
      covered: true,
      hull_lockable: true,
      maintained_by_turku: true,
    },
  },
};

const renderWithProviders = getRenderWithProviders({});

describe('<BicycleStandContent />', () => {
  it('should work', () => {
    const { container } = renderWithProviders(<BicycleStandContent {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  it('does show text correctly', () => {
    const { container } = renderWithProviders(<BicycleStandContent {...mockProps} />);

    const h3 = container.querySelector('h3');
    const p = container.querySelectorAll('p');
    expect(h3.textContent).toContain(mockProps.bicycleStand.name);
    expect(p[0].textContent).toContain('Malli: Testimalli');
    expect(p[1].textContent).toContain('Pyöräpaikkojen määrä: 10');
    expect(p[2].textContent).toContain('Pyörätelineiden määrä: 5');
    expect(p[3].textContent).toContain(finnishTranslations['mobilityPlatform.content.bicycleStands.covered']);
    expect(p[4].textContent).toContain(finnishTranslations['mobilityPlatform.content.bicycleStands.hullLockable']);
    expect(p[5].textContent).toContain(finnishTranslations['mobilityPlatform.content.bicycleStands.maintainedByTku']);
  });
});
