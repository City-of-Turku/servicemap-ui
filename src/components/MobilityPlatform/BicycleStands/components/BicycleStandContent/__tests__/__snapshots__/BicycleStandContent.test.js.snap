// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BicycleStandContent /> should work 1`] = `
<div>
  <div
    class="css-1ynyhby"
  >
    <div
      class="css-1dznbe2"
    >
      <h3
        class="MuiTypography-root MuiTypography-subtitle1 css-dlbwt1-MuiTypography-root"
      >
        Testinimi
      </h3>
    </div>
    <div>
      <div
        class="css-1fhgjcy"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
        >
          Malli: Testimalli
        </p>
      </div>
      <div
        class="css-1fhgjcy"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
        >
          Pyöräpaikkojen määrä: 10
        </p>
      </div>
      <div
        class="css-1fhgjcy"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
        >
          Pyörätelineiden määrä: 5
        </p>
      </div>
      <div
        class="css-1fhgjcy"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 css-1rexknm-MuiTypography-root"
        >
          Pyöräparkki on katettu
        </p>
      </div>
      <div
        class="css-1fhgjcy"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 css-1rexknm-MuiTypography-root"
        >
          Pyörän voi runkolukita
        </p>
      </div>
      <div
        class="css-1fhgjcy"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 css-1rexknm-MuiTypography-root"
        >
          Turun kaupungin ylläpitämä.
        </p>
      </div>
    </div>
  </div>
</div>
`;
